### PR TITLE
Fixed side panel color issues from CR 124

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/37444
fix https://github.com/brave/brave-browser/issues/37451
fix https://github.com/brave/brave-browser/issues/37442
fix https://github.com/brave/brave-browser/issues/37450

Screenshot 2024-04-10 163401
Screenshot 2024-04-10 163528
Screenshot 2024-04-10 163554
Screenshot 2024-04-10 173449

Calling `AddMaterialUiColorMixer()` fixed https://github.com/brave/brave-browser/issues/37450 also.
ToggleButton's color id(ex, `kColorToggleButtonThumbOn` is set from there)

Resolves

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See the linked issue

### CODE REVIEW

File: chromium_src/ui/color/color_mixers.cc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+bool GiveTrue() {
+  return true;
+}
+
+}  // namespace
+
+// AddMaterialUiColorMixer() should be called to get some colors(ex,
+// kColorToastBackground) at specific order in the AddColorMixers().
+// See the comments at ui::AddColorMixers().
+#define IsChromeRefresh2023 IsChromeRefresh2023() || GiveTrue
```
Comment (mkarolin) [inline chromium_src/ui/color/color_mixers.cc:21]: @simonhong, do you want to get rid of the change from https://github.com/brave/brave-core/pull/22320/commits/b39c27459f5c34d61a95019547ef676f8dc70e38 that calls `AddMaterialSidePanelColorMixer` from `browser/ui/color/brave_color_mixer.cc` and move the color changes  to the `browser/ui/color/brave_material_side_panel_color_mixer.cc` in here?

Comment (mkarolin) [reply chromium_src/ui/color/color_mixers.cc:21]: Ah, never mind, that on is actually called from `chrome/browser/ui/color/chrome_color_mixers.cc`, but I think we could move the colors here still?

Comment (simonhong) [reply chromium_src/ui/color/color_mixers.cc:21]: Yeah, addressed - https://github.com/brave/brave-core/pull/22996/commits/f8bd1ecee9caedfde632ccab3625f7e11bc27826

File: browser/ui/color/brave_color_mixer.cc [hunk 2]
```diff
@@ -236,16 +236,6 @@ void AddChromeLightThemeColorMixer(ui::ColorProvider* provider,
   mixer[kColorTabFocusRingActive] = {ui::kColorFocusableBorderFocused};
   mixer[kColorTabFocusRingInactive] = {ui::kColorFocusableBorderFocused};
 
-  // Sidebar tweaks
-  mixer[kColorSidePanelContentBackground] = {SK_ColorWHITE};
-  mixer[ui::kColorListItemUrlFaviconBackground] = {gfx::kGoogleGrey050};
-
-  // Combobox hover
-  mixer[ui::kColorComboboxInkDropHovered] = {ui::kColorSysStateHoverOnSubtle};
```
Comment (simonhong) [inline browser/ui/color/brave_color_mixer.cc:244]: It's just deleted as we're calling `AddMaterialUiColorMixer()`.

**Review by mkarolin — APPROVED** [review]
LGTM

**Pull request merged**